### ISSUE TITLE
Bug: axe-core does not flag nested interactive elements when a <button> is inside an <a>.

### ISSUE THREAD

**Issue by Gcamara14** [dequelabs/axe-core#4986]
### Product

axe-core

### Product Version

Axe 4.10.3

### Latest Version

- [x] I have tested the issue with the latest version of the product

#### **Expectation**

axe-core should report an accessibility violation when interactive elements are nested.

#### **Actual**

axe-core does not flag any issues for a `<button>` nested inside an `<a>`.

#### **How to Reproduce**

```html
<a href="#">
  <button>testing</button>
</a>
```

#### **Additional context**

Nested interactive controls are invalid HTML and can cause accessibility and usability issues, but this pattern is currently not detected. Screen readers struggle with it.

## blog post I wrote a few years ago:
- https://giovanicamara.com/blog/nested-buttons-and-links/

**Comment by straker** [comment]
Thanks for the issue. The reason axe-core does not catch this is because axe-core only focuses on valid HTML that could cause this problem (i.e. ARIA roles being nested within other ARIA roles or interactive elements). The provided HTLM is not valid HTML and so axe-core does not try to catch it for accessibility issues.

<img width="1061" height="803" alt="Screenshot of HTML validator showing an anchor nested inside a button is invalid HTML" src="https://github.com/user-attachments/assets/550bbb3e-14b9-4319-a9b6-3846ec06c93c" />
